src/ape/utils/os.py

```diff
@@ -139,3 +139,18 @@ def __exit__(self, *exc):
         for path in self.exclude:
             if path not in sys.path:
                 sys.path.append(path)
+
+
+def get_full_extension(path: Path) -> str:
+    """
+    For a path like ``Path("Contract.t.sol")``,
+    returns ``.t.sol``, unlike the regular Path
+    property ``.suffix`` which returns ``.sol``.
+    """
+    if path.is_dir():
+        return ""
+
+    parts = path.name.split(".")
+    start_idx = 2 if path.name.startswith(".") else 1
+    suffix = ".".join(parts[start_idx:])
```

Comment:
this doesn't raise when it's just `.something`?

Reply:
It would fail the test I added if it did! But just for funzies, I will execute each line individually:

```
In [2]: path = Path(".something")

In [3]: parts = path.name.split(".")

In [4]: parts
Out[4]: ['', 'something']

In [5]: start_idx = 2 if path.name.startswith(".") else 1

In [6]: start_idx
Out[6]: 2

In [7]: suffix = ".".join(parts[start_idx:])

In [8]: suffix
Out[8]: ''

In [9]: f".{suffix}" if suffix and f".{suffix}" != f"{path.name}" else ""
Out[9]: ''
```

**Note**: In python, when using slices, the indices are allowed to exceed the last index in a sequence. For example:

```python
In [13]: my_lst[:9999]
Out[13]: [1, 2, 3]

In [14]: my_lst[9999:]
Out[14]: []
```

It is the design of slices, to make them more convenient.

Reply:
All that being said, I am happy to put in an index check for the sake of readability.

Reply:
> **Note**: In python, when using slices, the indices are allowed to exceed the last index in a sequence. For example:
> 
> ```python
> In [13]: my_lst[:9999]
> Out[13]: [1, 2, 3]
> 
> In [14]: my_lst[9999:]
> Out[14]: []
> ```
> 
> It is the design of slices, to make them more convenient.

TIL

Reply:
> All that being said, I am happy to put in an index check for the sake of readability. Shall I?

No we are good! I just didn't understand that bit

Maybe add a comment?